src/main/scala/com/fulcrumgenomics/sv/tools/SvPileup.scala

```diff
@@ -121,157 +120,166 @@ class SvPileup
-      EvidenceType.values.zip(this.counts(bp))
-    }
+  /**
+   * Performs filtering on a Template to remove primary records that are unmapped and then remove
+   * supplementary records if there is no matching primary record retained.  If neither primary
+   * record is retained, returns None, else returns Some(Template).
+   */
+  def filterTemplate(t: Template,
+                     minPrimaryMapq: Int,
+                     minSupplementaryMapq: Int): Option[Template] = {
+    val r1PrimaryOk = t.r1.exists(r => r.mapped && r.mapq >= minPrimaryMapq)
+    val r2PrimaryOk = t.r2.exists(r => r.mapped && r.mapq >= minPrimaryMapq)
+
+    if (!r1PrimaryOk && !r2PrimaryOk) None else Some(
+      Template(
+        r1 = if (r1PrimaryOk) t.r1 else None,
+        r2 = if (r2PrimaryOk) t.r2 else None,
+        r1Supplementals = if (r1PrimaryOk) t.r1Supplementals.filter(_.mapq >= minSupplementaryMapq) else Nil,
+        r2Supplementals = if (r2PrimaryOk) t.r2Supplementals.filter(_.mapq >= minSupplementaryMapq) else Nil,
+      )
+    )
   }
 
+
   /** Finds the breakpoints for the given template.
    *
    * @param template the template to examine
+   * @param maxWithinReadDistance the maximum distance between two adjacent split read mappings before calling a breakpoint
    * @param maxReadPairInnerDistance the maximum inner distance between R1 and R2 before calling a breakpoint
-   * @param maxAlignedSegmentInnerDistance the maximum distance between two adjacent split read mappings before calling
-   *                                       a breakpoint
-   * @param minSupplementaryMappingQuality the minimum mapping quailty to keep a supplementary alignment
    * @param minUniqueBasesToAdd the minimum newly covered bases to keep a supplementary alignment when iteratively
    *                            adding them.
    */
   def findBreakpoints(template: Template,
+                      maxWithinReadDistance: Int,
                       maxReadPairInnerDistance: Int,
-                      maxAlignedSegmentInnerDistance: Int,
-                      minSupplementaryMappingQuality: Int,
                       minUniqueBasesToAdd: Int,
                      ): IndexedSeq[BreakpointEvidence] = {
-    val segments = AlignedSegment.segmentsFrom(
-      template                       = template,
-      minSupplementaryMappingQuality = minSupplementaryMappingQuality,
-      minUniqueBasesToAdd            = minUniqueBasesToAdd
-    )
+    val segments = AlignedSegment.segmentsFrom(template, minUniqueBasesToAdd = minUniqueBasesToAdd)
+
+    segments.length match {
+      case 0 | 1 =>
+        NoBreakpoints
+      case 2     =>
+        // Special case for 2 since most templates will generate two segments and we'd like it to be efficient
+        val bp = findBreakpoint(segments.head, segments.last, maxWithinReadDistance, maxReadPairInnerDistance)
+        if (bp.isEmpty) NoBreakpoints else bp.toIndexedSeq
```

Comment:
Isn't `.toIndexedSeq` redundant?
```suggestion
        if (bp.isEmpty) NoBreakpoints else bp
```

Reply:
`bp` is an Option which does not implement `IndexedSeq`.